internal/core/data/event_test.go

```diff
@@ -347,6 +636,60 @@ func newMockDeviceClient() *mocks.DeviceClient {
 	return client
 }
 
+func newMockDb() interfaces.DBClient {
+	db := &dbMock.DBClient{}
```

Comment:
Question -- Did you notice that in the reset() function in this file, there's a different usage of the dbClient?

`dbClient = &memory.MemDB{}`

I would encourage you to look at that provider and try to use it for your tests. Every test calls reset() to reinitialize state within the MemDB.

Reply:
Yes. I wanted to be able to get coverage on the error handling parts of the DB interactions, and on some of these memory DB function calls there's so little actually working that it's impossible to generate an error. I probably have too much reliance on mock calls versus the stub memory DB, but it's so much easier to just have the mock throw an error and see how event deals with it than try to organically generate the error that I thought this was worth it.

Reply:
I'm going to leave this out here overnight and see if either @feclare or @jduranf has any comment on this topic.